### PR TITLE
Reorganize the repo structure

### PR DESCRIPTION
**Pull request by mengqiy**
Moved contrib functions to functions/contrib.
Created subdirectories mutators, validators and contrib in examples/

closes: https://github.com/GoogleContainerTools/kpt/issues/1368

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,11 +38,14 @@ test: unit-test e2e-test ## Run all unit tests and e2e tests
 check-licenses:
 	cd functions/ts && $(MAKE) check-licenses
 	cd functions/go && $(MAKE) check-licenses
+	cd functions/contrib/go && $(MAKE) check-licenses
```
Comment (Shell32-Natsu) [inline Makefile:41]: Why isn't `contrib/ts` here?

File: functions/contrib/ts/helm-inflator/src/gen/io.k8s.api.admissionregistration.v1beta1.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import { KubernetesObject } from 'kpt-functions';
-import * as apisMetaV1 from './io.k8s.apimachinery.pkg.apis.meta.v1';
+import { KubernetesObject } from "kpt-functions";
```
Comment (Shell32-Natsu) [inline functions/contrib/ts/helm-inflator/src/gen/io.k8s.api.admissionregistration.v1beta1.ts:1]: I suppose these quotes should have already checked by prettier. Why are they changed?

Comment (mengqiy) [reply functions/contrib/ts/helm-inflator/src/gen/io.k8s.api.admissionregistration.v1beta1.ts]: I didn't copy over `functions/contrib/.prettierrc.json`. It uses single quote again.

File: Makefile [hunk 3]
```diff
@@ -37,11 +38,14 @@ test: unit-test e2e-test ## Run all unit tests and e2e tests
 check-licenses:
 	cd functions/ts && $(MAKE) check-licenses
 	cd functions/go && $(MAKE) check-licenses
+	cd functions/contrib/ts && $(MAKE) check-licenses
```
Comment (mengqiy) [inline Makefile:41]: @Shell32-Natsu Re. your earlier question, we only have functions written in TS in the contrib dir.

**Review by Shell32-Natsu — COMMENTED** [review]
I am still thinking using `docs` make this confusing. These files are not docs.

**Comment by mengqiy** [comment]
Due to nature of self-contained packages, we will have kpt packages and markdown files mixed together.
But we are going to render these markdown files in our catalog site.
If using `docs` as directory name is confusing, I can change it back to `examples`.

**Review by Shell32-Natsu — APPROVED** [review]
lgtm

**Comment by mengqiy** [comment]
cc: @etefera

**Pull request merged**